src/ReactWrapper.jsx

```diff
@@ -243,11 +277,14 @@ class ReactWrapper {
    * @param {Function} cb - callback function
    * @returns {ReactWrapper}
    */
-  setProps(props, callback = undefined) {
+  setProps(props, callback = noop) {
     if (this.root !== this) {
       throw new Error('ReactWrapper::setProps() can only be called on the root');
     }
-    this.component.setChildProps(props, callback);
+    this.component.setChildProps(props, () => {
+      this.update();
+      callback();
```

Comment:
Sure, I guess - it just adds function call overhead in the case when nothing is provided, so i'm not sure why this is better than defaulting to `undefined`.